### PR TITLE
Remove optional step at knative/client repo

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes

This step is no longer needed. We leverage auto generated release notes from PRs.

- Remove optional step at knative/client repo

/cc @knative/knative-release-leads

### CODE REVIEW

**Comment by kvmware** [comment]
Also, don't know if this is mentioned in the PROCEDURE or TIMELINE docs but should be removed there too if so.

**Comment by dprotaso** [comment]
FYI - needs a rebase

File: PROCEDURES.md [hunk 1]
```diff
@@ -64,13 +64,6 @@ If the releasability test reports a NO-GO on a repo where it was previously pass
 ## Cut a branch
 tl;dr: cut a release branch from `main`.
 
-### Exceptions
-For some repositories some extra manual validation and updates need to be performed before the release branch is cut:
-
-- `knative/client`
-  - (optional) Check that [CHANGELOG.adoc](https://github.com/knative/client/blob/main/CHANGELOG.adoc) contains a section about the release, i.e. the top-level "(Unreleased)" section should be changed to point to the upcoming release version and number. It's not critical if the changelog is aligned after the release in retrospective.
-  - If the validation fails, the fix should be trivial and could be either performed by the release leads or the client team.
-
```
Comment (dsimansk) [inline PROCEDURES.md:73]: Note: removing the whole Exceptions section in a omit empty fashion. 

@dprotaso @kvmware
